### PR TITLE
[GCI-142] Replace code with computeIfAbsent()

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
Issue: https://issues.openmrs.org/browse/GCI-142

computeIfAbsent() is great Stream API method that checks for element's existence in Collection, and if it doesn't exist, does some work. I think it is well explained here https://stackoverflow.com/questions/19278443/how-do-i-use-the-new-computeifabsent-function/19283210#19283210

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/ModuleFactory.java [hunk 1]
```diff
@@ -704,12 +704,8 @@ public static Module startModuleInternal(Module module, boolean isOpenmrsStartup
 				for (Extension ext : module.getExtensions()) {
 					
 					String extId = ext.getExtensionId();
-					List<Extension> tmpExtensions = moduleExtensionMap.get(extId);
-					if (tmpExtensions == null) {
-						tmpExtensions = new ArrayList<>();
-						moduleExtensionMap.put(extId, tmpExtensions);
-					}
-
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/ModuleFactory.java]: Did you intentionally replace ArrayList with Vector?

File: api/src/main/java/org/openmrs/module/ModuleFactory.java [hunk 2]
```diff
@@ -727,11 +723,8 @@ public int compare(Extension e1, Extension e2) {
 					sortedModuleExtensions.sort(sortOrder);
 					
 					// Get existing extensions, and append the ones from the new module
-					List<Extension> extensions = getExtensionMap().get(moduleExtensionEntry.getKey());
-					if (extensions == null) {
-						extensions = new ArrayList<>();
-						getExtensionMap().put(moduleExtensionEntry.getKey(), extensions);
-					}
+					List<Extension> extensions = getExtensionMap()
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/ModuleFactory.java]: Did you intentionally replace ArrayList with Vector?

File: api/src/main/java/org/openmrs/module/ModuleFactory.java (hunk 1 above)
Comment (BartlomiejRasztabiga) [reply api/src/main/java/org/openmrs/module/ModuleFactory.java]: Fixed :)

**Comment by dkayiwa** [comment]
Thanks @BartlomiejRasztabiga 😊

**Pull request merged**